### PR TITLE
Referrals beta

### PR DESCRIPTION
**Pull request by ecotopian**

### CODE REVIEW

File: Client/Ecosia/Extensions/Referrals+BuildType.swift [hunk 1]
```diff
@@ -10,8 +10,6 @@ import Shared
 extension Referrals {
     static var isEnabledForBuild: Bool {
         switch AppConstants.BuildChannel {
-        case .release:
```
Comment (vauxhall) [inline Client/Ecosia/Extensions/Referrals+BuildType.swift:13]: I wonder if we need this `enum` at all anymore.

**Review by vauxhall — APPROVED** [review]
Looks great, just added a comment 👍

File: Client/Ecosia/UI/MyImpact/EcosiaHome.swift [hunk 2]
```diff
@@ -204,6 +204,11 @@ final class EcosiaHome: UICollectionViewController, UICollectionViewDelegateFlow
         super.viewWillAppear(animated)
         news.load(session: .shared, force: items.isEmpty)
         Analytics.shared.navigation(.view, label: .home)
+
+        if Referrals.isActive {
+            referrals.refresh(force: true)
```
Comment (vauxhall) [inline Client/Ecosia/UI/MyImpact/EcosiaHome.swift:209]: @ecotopian we are always forcing a refresh?

File: Client/Ecosia/Extensions/Referrals+BuildType.swift (hunk 1 above)
Comment (ecotopian) [reply Client/Ecosia/Extensions/Referrals+BuildType.swift:13]: yes - agree. I somehow felt we might revive it but it's better to make it clean in the first place

File: Client/Ecosia/UI/MyImpact/EcosiaHome.swift (hunk 2 above)
Comment (ecotopian) [reply Client/Ecosia/UI/MyImpact/EcosiaHome.swift:209]: @vauxhall no we don't. In FirefoxHomeViewController (NTP) on viewWillAppear we do a refresh without force

File: Client/Frontend/Home/FirefoxHomeViewController.swift [hunk 3]
```diff
@@ -351,15 +351,16 @@ class FirefoxHomeViewController: UICollectionViewController, HomePanel, FeatureF
...
-            referrals.startSilentRefresh()
-        }
     }
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         reloadAll()
 
+        if Referrals.isActive {
+            referrals.refresh()
```
Comment (ecotopian) [inline Client/Frontend/Home/FirefoxHomeViewController.swift:361]: @vauxhall here we do the "regular" refresh which will be throttled by 1 hour logic

File: Client/Ecosia/Extensions/Referrals+BuildType.swift (hunk 1 above)
Comment (ecotopian) [reply Client/Ecosia/Extensions/Referrals+BuildType.swift]: @vauxhall FYI: I'll keep the logic for now for version 6.0.3 (which shall not have referrals) and we'll remove the enum for 6.1.0

**Pull request merged**